From penberg@cs.helsinki.fi  Wed Nov 12 13:55:21 2008
From: Pekka Enberg <penberg@cs.helsinki.fi>
Date: Thu, 30 Oct 2008 19:04:51 +0200
Subject: Staging: w35und: make functions local to mlmetxrx.c static
To: greg@kroah.com
Cc: Pekka Enberg <penberg@cs.helsinki.fi>, Pavel Machek <pavel@suse.cz>
Message-ID: <1225386296-19632-2-git-send-email-penberg@cs.helsinki.fi>


Convert functions that don't have external references to static and remove the
ones that are not used at all.

Acked-by: Pavel Machek <pavel@suse.cz>
Signed-off-by: Pekka Enberg <penberg@cs.helsinki.fi>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/winbond/mlmetxrx.c   |   55 -----------------------------------
 drivers/staging/winbond/mlmetxrx_f.h |    9 -----
 2 files changed, 1 insertion(+), 63 deletions(-)

--- a/drivers/staging/winbond/mlmetxrx.c
+++ b/drivers/staging/winbond/mlmetxrx.c
@@ -19,59 +19,6 @@
 
 #include "mds_f.h"
 
-void MLMEResetTxRx(struct wbsoft_priv * adapter)
-{
-	s32     i;
-
-	// Reset the interface between MDS and MLME
-	for (i = 0; i < MAX_NUM_TX_MMPDU; i++)
-		adapter->sMlmeFrame.TxMMPDUInUse[i] = false;
-	for (i = 0; i < MAX_NUM_RX_MMPDU; i++)
-		adapter->sMlmeFrame.SaveRxBufSlotInUse[i] = false;
-
-	adapter->sMlmeFrame.wNumRxMMPDUInMLME   = 0;
-	adapter->sMlmeFrame.wNumRxMMPDUDiscarded = 0;
-	adapter->sMlmeFrame.wNumRxMMPDU          = 0;
-	adapter->sMlmeFrame.wNumTxMMPDUDiscarded = 0;
-	adapter->sMlmeFrame.wNumTxMMPDU          = 0;
-	adapter->sLocalPara.boCCAbusy    = false;
-	adapter->sLocalPara.iPowerSaveMode     = PWR_ACTIVE;     // Power active
-}
-
-//=============================================================================
-//	Function:
-//    MLMEGetMMPDUBuffer()
-//
-//	Description:
-//    Return the pointer to an available data buffer with
-//    the size MAX_MMPDU_SIZE for a MMPDU.
-//
-//  Arguments:
-//    adapter   - pointer to the miniport adapter context.
-//
-//	Return value:
-//    NULL     : No available data buffer available
-//    Otherwise: Pointer to the data buffer
-//=============================================================================
-
-/* FIXME: Should this just be replaced with kmalloc() and kfree()? */
-u8 *MLMEGetMMPDUBuffer(struct wbsoft_priv * adapter)
-{
-	s32 i;
-	u8 *returnVal;
-
-	for (i = 0; i< MAX_NUM_TX_MMPDU; i++) {
-		if (adapter->sMlmeFrame.TxMMPDUInUse[i] == false)
-			break;
-	}
-	if (i >= MAX_NUM_TX_MMPDU) return NULL;
-
-	returnVal = (u8 *)&(adapter->sMlmeFrame.TxMMPDU[i]);
-	adapter->sMlmeFrame.TxMMPDUInUse[i] = true;
-
-	return returnVal;
-}
-
 //=============================================================================
 u8 MLMESendFrame(struct wbsoft_priv * adapter, u8 *pMMPDU, u16 len, u8 DataType)
 /*	DataType : FRAME_TYPE_802_11_MANAGEMENT, FRAME_TYPE_802_11_MANAGEMENT_CHALLENGE,
@@ -109,7 +56,7 @@ void MLME_GetNextPacket(struct wbsoft_pr
 	desc->Type = adapter->sMlmeFrame.DataType;
 }
 
-void MLMEfreeMMPDUBuffer(struct wbsoft_priv * adapter, s8 *pData)
+static void MLMEfreeMMPDUBuffer(struct wbsoft_priv *adapter, s8 *pData)
 {
 	int i;
 
--- a/drivers/staging/winbond/mlmetxrx_f.h
+++ b/drivers/staging/winbond/mlmetxrx_f.h
@@ -15,15 +15,6 @@ MLMEProcThread(
      struct wbsoft_priv *    adapter
 	);
 
-void MLMEResetTxRx( struct wbsoft_priv * adapter);
-
-u8 *
-MLMEGetMMPDUBuffer(
-     struct wbsoft_priv *    adapter
-   );
-
-void MLMEfreeMMPDUBuffer( struct wbsoft_priv * adapter,  s8 * pData);
-
 void MLME_GetNextPacket(  struct wbsoft_priv * adapter,  PDESCRIPTOR pDes );
 u8 MLMESendFrame( struct wbsoft_priv * adapter,
 					u8	*pMMPDU,
